openshift_pool/openshift/stack.py

```diff
@@ -267,8 +288,37 @@ def hosts_data(self):
 
     def get_connection_statuses(self):
         """Return a LUT that contains each node and its connectivity by the domain"""
+
         out = {}
+
+        def test_ping(hostname):
+            self.log.debug(f"testing ping connectivity to host {hostname}")
+            result = subprocess.run(['ping', '-c', '1', hostname], stdout = subprocess.PIPE)
+            self.log.debug(f"[STDOUT]: {result.stdout}")
+            self.log.debug(f"[STDERR]: {result.stderr}")
+            self.log.debug(f"[RC]: {result.returncode}")
+
+            return bool(not result.returncode)
+
+        def test_ssh(hostname, username, password):
+            try:
+                self.log.debug(f"testing ssh connectivity to host {hostname} (creds: {username}\{password})")
+                is_ssh = False
+                client = paramiko.SSHClient()
+                client.load_system_host_keys()
+                client.set_missing_host_key_policy(paramiko.MissingHostKeyPolicy())
+                client.connect(hostname,username=username,password=password)
+                is_ssh = client.get_transport().isAlive()
+            except Exception:
+                pass
+            finally:
+                client.close()
```

Comment:
if we use Paramiko here, we would not have to close the ssh client. 

https://github.com/paramiko/paramiko/blob/master/paramiko/util.py#L295

`with Paramiko.SSHClient(<args>) as client:`